sklearn/impute/_base.py

```diff
@@ -465,11 +465,45 @@ def transform(self, X):
             n_missing = np.sum(mask, axis=0)
             values = np.repeat(valid_statistics, n_missing)
             coordinates = np.where(mask.transpose())[::-1]
-
             X[coordinates] = values
 
         return super()._concatenate_indicator(X, X_indicator)
 
+    def inverse_transform(self, X):
+        """Convert the data back to the original representation.
+
+        Inverts the ``fit_transform`` operation performed on an array.
+        This operation can only be performed after :class:`SimpleImputer` is
+        instantiated with ``add_indicator`` parameter set to `True`.
+
+        Note that ``inverse_transform`` can only invert the transform in
+        features that have binary indicators for missing values. If a feature
+        has no missing values at ``fit`` time, the feature won't have a binary
+        indicator, and the imputation done at ``transform`` time won't be
+        inverted.
+
+        Parameters
+        ----------
+        X : array-like, \
+            shape (n_samples, n_features + missing_feature_count)
+            The imputed data to be reverted to original data. It has to be
+            an augmented array of imputed data and the missing indicator mask.
+        """
+        check_is_fitted(self)
```

Comment:
We need to raise an error if `self.add_indicator=False`

Reply:
Raising an error now, and added a test case for it